actions/upload_sessions.go

```diff
@@ -175,11 +204,15 @@ func (usr *UploadSessionResource) Update(c buffalo.Context) error {
 			key := sqlWhereForGenesisHashAndChunkIdx(uploadSession.GenesisHash, chunkIdx)
 			sqlWhereClosures = append(sqlWhereClosures, key)
 			chunksMap[key] = chunk
+			minChunkIdx = math.Min(minChunkIdx, float64(chunkIdx))
+			maxChunkIdx = math.Max(maxChunkIdx, float64(chunkIdx))
 		}
 
 		var dms []models.DataMap
-		rawQuery := fmt.Sprintf("SELECT * from data_maps WHERE %s", strings.Join(sqlWhereClosures, " OR "))
-		err := models.DB.RawQuery(rawQuery).All(&dms)
+		//rawQuery := fmt.Sprintf("SELECT * from data_maps WHERE %s", strings.Join(sqlWhereClosures, " OR "))
+		err := models.DB.RawQuery(
+			"SELECT * from data_maps WHERE genesis_hash = ? AND chunk_idx >= ? AND chunk_idx <= ?",
+			uploadSession.GenesisHash, minChunkIdx, maxChunkIdx).All(&dms)
```

Comment:
This may not be the best way to handle this.  But this line is where we were getting the "busy buffer" error before.  Making the query smaller seems to have fixed it.  

As long as we continue to send the chunks in contiguous batches and don't send them out of order or something, this should work.